File: physical/walk.js
// function retrieveName (e) {
//     e.preventDefault()

//     let username = document.getElementById('uname').value

//     fetch(`http://127.0.0.1:5000/${username}`)
//     .then((res)=> res.json())
//     .then((data)=> {
//         document.getElementById('output').innerHTML = `<li class="list-group-item">"Login successfully!"</li> `
//     })
// }


document.getElementById('walkData').addEventListener('submit', walkData)

function walkData (e) {
    e.preventDefault()

    let username = document.getElementById('uname').value
    let duration = document.getElementById('q1').value
    let distance = document.getElementById('q2').value
    let preference = document.getElementById('q3').value
    
   
    fetch('http://127.0.0.1:5000/walkData', {
        method : 'POST',
        headers : {
            'Content-Type' : 'application/json'
        },
        body : JSON.stringify({
            'uname' : username,
            'q1' : duration,
            'q2' : distance,
            'q3' : preference
        })
    })
    .then((response) => response.json())
    .then((data) => console.log(data))
}

File: physical/walk.py
from flask import Flask, jsonify, request, redirect
from flask.helpers import url_for
from flask_pymongo import PyMongo
from flask_cors import CORS, cross_origin

walk = Flask(__name__)
cors = CORS(walk)
walk.config['MONGO_URI'] = 'mongodb+srv://HealthPal:HealthPal@cluster0.6hpjx.mongodb.net/HealthPal?retryWrites=false&w=majority'
walk.config['CORS_Headers'] = 'Content-Type'
mongo = PyMongo(walk)

@walk.route('/walkData', methods = ['POST'])
def postData():
    currentCollection = mongo.db.walk
    username = request.json['uname']
    duration = request.json['q1']
    distance = request.json['q2']
    preference = request.json['q3']
    currentCollection.insert_one({'uname' : username, 'q1' : duration, 'q2' : distance, 'q3' : preference})
    return jsonify({'uname' : username, 'q1' : duration})


if __name__ == '__main__':
    walk.run(debug = True)